src/psygnal/_group_descriptor.py

```diff
@@ -656,3 +705,50 @@ def _create_group(self, owner: type) -> type[SignalGroup]:
 
         self._do_patch_setattr(owner, with_aliases=bool(Group._psygnal_aliases))
         return Group
+
+
+@dataclass
+class Validator:
+    """Annotated metadatax that marks a function validates a value before setting.
```

Comment:
```suggestion
    """Annotated metadata marking that a function validates a value before setting.
```
not sure if this captures the intent